src/compact_log.rs

```diff
@@ -102,6 +105,14 @@ impl core::fmt::Debug for CompactLogItem {
     }
 }
 
+impl race_buffer::Entry for CompactLogItem {
+    const NIL_VAL: Self = CompactLogItem::event(EventId::EVENT_NIL_VALUE);
+
+    fn is_prefix(&self) -> bool {
```

Comment:
I'm worried about naive, log-context-free interpretation of this method.  The reason is that both the "count" and the "payload" log items may very well have those bits set.  

Without correct alignment of log interpretation (e.g. awareness of log item position relative to a stable start-point), it's possible to mistake one of the payload log items for one of the "prefixy" items.

Reply:
I was actually not aware that the count and payload items were allowed to use those bits, but the RaceBuffer guarantees that the reader never reads the value after a prefix if the prefix is missed. I will do more model checking to ensure that this doesn't cause problems.